api/services/registry/server.go

```diff
@@ -106,3 +109,38 @@ func (s *Server) Proxy(stream Registry_ProxyServer) error {
 
 	return nil
 }
+
+// CopyWithDeadline copies data from a net.Conn using a read deadline. The
+// process will fail with a timeout error if no data is read for the defined
+// period.
+func CopyWithDeadline(conn net.Conn, w io.Writer) (int64, error) {
+	var t int64
+	for {
+		err := conn.SetReadDeadline(time.Now().Add(readDeadline))
+		if err != nil {
+			return 0, err
```

Comment:
nit: we should return the number of bytes successfully written here (and in other cases we return errors) rather than 0 -- this gives the caller of this function an opportunity to handle errors more accurately. I see the current caller doesn't use it, but this is a bug waiting to happen if down the road a different caller makes use of this function.